autorest/azure/auth/authfile.go

```diff
@@ -0,0 +1,103 @@
+package auth
+
+import (
+	"bytes"
+	"encoding/binary"
+	"encoding/json"
+	"errors"
+	"io/ioutil"
+	"os"
+	"unicode/utf16"
+
+	"github.com/Azure/go-autorest/autorest"
+	"github.com/Azure/go-autorest/autorest/adal"
+	"github.com/dimchansky/utfbom"
+)
+
+var pcEndpoints = map[string]string{
+	"managementEndpointUrl":          "https://management.core.windows.net",
+	"resourceManagerEndpointUrl":     "https://management.azure.com",
+	"activeDirectoryEndpointUrl":     "https://login.microsoftonline.com",
+	"galleryEndpointUrl":             "https://gallery.azure.com",
+	"activeDirectoryGraphResourceId": "https://graph.windows.net",
+}
+
+// Auth includes authentication details for ARM clients
+type Auth struct {
```

Comment:
I think this type (or actually I think you should swap it out for a new type based on what the CLI did, see below) should implement this interface: https://github.com/Azure/go-autorest/blob/faeb847c8586092daaa111b17cc4e6c792fd3a87/autorest/authorization.go#L21